setup.py

```diff
@@ -65,6 +70,7 @@
     "FIXED_INTERP": "yt/utilities/lib/fixed_interpolator.cpp",
     "ARTIO_SOURCE": glob.glob("yt/frontends/artio/artio_headers/*.c"),
     "CPP14_FLAG": CPP14_FLAG,
+    "CPP03_FLAG": CPP03_FLAG,
```

Comment:
If possible, build all binaries with the same C++ standard. Especially between C++03 and C++11 were ABI breaks in the STL and with MSVC the math libs and STL will also cause problems.

Reply:
I believe it is fine to replace the `CPP03_flag` with the `CPP14_flag` in the two files that use it. I added commit suggestions where relevant.
```suggestion
```